v2/kafka-to-gcs/src/main/java/com/google/cloud/teleport/v2/templates/KafkaToGCS.java

```diff
@@ -0,0 +1,264 @@
+ *           "outputFilenamePrefix":"output",
+ *           "windowDuration":"5m",
+ *           "numShards":"5"
+ *        }
+ *       }
+ *      '
+ * </pre>
+ */
+public class KafkaToGCS implements WriteToGCS {
+
+  /* Logger for class. */
+  private static final Logger LOG = LoggerFactory.getLogger(KafkaToGCS.class);
+
+  /**
+   * The {@link KafkaToGCSOptions} class provides the custom execution options passed by the
+   * executor at the command-line.
+   */
+  public interface KafkaToGCSOptions extends PipelineOptions {
+    @Description("Kafka Bootstrap Servers. ")
+    @Required
+    String getBootstrapServers();
+
+    void setBootstrapServers(String bootstrapServers);
+
+    @Description("Kafka topic to read the input from. ")
+    @Required
+    String getInputTopics();
+
+    void setInputTopics(String inputTopics);
+
+    @Description("The directory to output files to. Must end with a slash. ")
+    @Required
+    String getOutputDirectory();
+
+    void setOutputDirectory(String outputDirectory);
+
+    @Description(
+        "The format of the file to write to. Text file is the Default File Format. "
+            + "Allowed formats are: "
+            + "TEXT, AVRO, PARQUET")
+    @Default.String("text")
+    String getOutputFileFormat();
+
+    void setOutputFileFormat(String outputFileFormat);
+
+    @Description(
+        "The filename prefix of the files to write to. Default file prefix is set to \"output\". ")
+    @Default.String("output")
+    String getOutputFilenamePrefix();
+
+    void setOutputFilenamePrefix(String outputFilenamePrefix);
+
+    @Description(
+        "The window duration in which data will be written. Defaults to 5m. "
+            + "Allowed formats are: "
+            + "Ns (for seconds, example: 5s), "
+            + "Nm (for minutes, example: 12m), "
+            + "Nh (for hours, example: 2h).")
+    @Default.String("5m")
+    String getWindowDuration();
+
+    void setWindowDuration(String windowDuration);
+
+    @Description(
+        "The maximum number of output shards produced when writing. Default set to 0 shards. ")
+    @Default.Integer(0)
+    Integer getNumShards();
+
+    void setNumShards(Integer numShards);
+  }
+
+  /**
+   * Main entry point for pipeline execution.
+   *
+   * @param args Command line arguments to the pipeline.
+   */
+  public static void main(String[] args) {
+    KafkaToGCSOptions options =
+        PipelineOptionsFactory.fromArgs(args).withValidation().as(KafkaToGCSOptions.class);
+
+    run(options);
+  }
+
+  /**
+   * Runs a pipeline which reads message from Kafka and writes it to GCS.
+   *
+   * @param options arguments to the pipeline
+   */
+  public static PipelineResult run(KafkaToGCSOptions options) {
+
+    String outputFileFormat = options.getOutputFileFormat().toUpperCase();
+    LOG.info("Requested File Format is " + outputFileFormat);
+
+    // Set the suffix for the file(s) to be stored.
+    WriteToGCS.setFileFormatSuffix();

```

Comment:
Why is this needed? There is nothing additional determined at runtime. This can be static